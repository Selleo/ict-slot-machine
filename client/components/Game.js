import React from 'react'
import Roller from './Roller'
import socketIOClient from 'socket.io-client'
import { resultResponse } from '../modules/result'
import { ToastContainer, toast } from 'react-toastify'

const ROLLERS = ['left', 'center', 'right']

export default class Game extends React.Component {
  constructor(props) {
    super(props)

    this.state = {}
    this.socket = socketIOClient('http://localhost:3000')

    ROLLERS.forEach(roller => {
      this[roller] = React.createRef()
    })
  }

  componentDidMount() {
    this.socket.on('SPIN_REQUEST', forcedSpinTo => this._spinMachine(forcedSpinTo))
    this.socket.on('NOTIFY', (type, message) => toast[type](message))
  }

  _spinMachine(forcedSpinTo) {
    const spins = ROLLERS.map(roller => this[roller].current.spin(this.state[roller], forcedSpinTo))

    Promise.all(spins).then(results => {
      const state = Object.values(results).reduce((a, v) => ({ ...a, ...v }), {})
      this.setState(state)

      this.socket.emit('SPIN_ENDED', resultResponse(state))
    })
  }

  _createRollers() {
    return ROLLERS.map(roller => (
      <Roller
        key={roller}
        ref={this[roller]}
        position={roller}
        height={170}
        width={210}
        count={10}
      />
    ))
  }

  render() {
    return (
      <div className="rollers">
        <div style={{background:'hotpink',position:'absolute',zIndex:'1333',top:'50px',left: '50%'}}><button onClick={()=>this.socket.emit('SPIN_REQUEST')}>test</button></div>
        <ToastContainer autoClose={5000} position={toast.POSITION.TOP_CENTER} />
        <div className="overlay" />
        {this._createRollers()}
      </div>
    )
  }
}
