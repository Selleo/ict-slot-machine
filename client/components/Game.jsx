import React from 'react'
import Roller from './Roller'
import socketIOClient from 'socket.io-client'
import { logResult, resultResponse } from '../modules/result'
import { ToastContainer, toast } from 'react-toastify'

const ROLLERS = ['left', 'center', 'right']

export default class Game extends React.Component {
  constructor(props) {
    super(props)

    this.state = {}
    this.socket = socketIOClient('http://localhost:3000')

    ROLLERS.forEach(roller => {
      this[roller] = React.createRef()
    })
  }

  componentDidMount() {
    this.socket.on('SPIN_REQUEST', forcedSpinTo => this._spinMachine(forcedSpinTo))
    this.socket.on('NOTIFY', (type, message) => toast[type](message))

    // toast.success("Success Notification!", {
    //   position: toast.POSITION.TOP_CENTER
    // });
  }

  _spinMachine(forcedSpinTo) {
    const spins = ROLLERS.map(roller => this[roller].current.spin(this.state[roller], forcedSpinTo))

    Promise.all(spins).then(results => {
      const state = Object.values(results).reduce((a, v) => ({ ...a, ...v }), {})
      this.setState(state)

      this.socket.emit('SPIN_ENDED', resultResponse(state))
    })
  }

  _createRollers() {
    return ROLLERS.map(roller => (
      <Roller
        key={roller}
        ref={this[roller]}
        position={roller}
        height={170}
        width={210}
        count={10}
      />
    ))
  }

  render() {
    return (
      <div className="rollers">
        <ToastContainer autoClose={3000} position={toast.POSITION.TOP_CENTER} />
        <div className="overlay" />
        {this._createRollers()}
      </div>
    )
  }
}
