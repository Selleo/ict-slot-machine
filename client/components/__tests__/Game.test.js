import React from 'react'
import renderer from 'react-test-renderer'
import Game from '../Game'

describe('<Game />', () => {
  const defaultProps = {}
  const wrapper = renderer.create(<Game {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
