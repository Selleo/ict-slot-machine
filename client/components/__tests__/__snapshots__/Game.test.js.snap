// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game /> render 1`] = `
<div
  className="rollers"
>
  <div
    className="Toastify"
  />
  <div
    className="overlay"
  />
  <div
    className="roller-wrapper left"
  >
    <div
      className="styles-roller-left"
    />
    <div
      className="roller left"
      style={
        Object {
          "height": "170px",
          "transformOrigin": "0 50% -261.6031006598966px",
          "width": "210px",
        }
      }
    >
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(0px) translateY(0px) rotateX(0deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486021px) translateY(153.76644452187057px) rotateX(-36deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.7632967748085px) translateY(248.79933356561165px) rotateX(-72deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.44290454498463px) translateY(248.79933356561168px) rotateX(-108deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.24445487493296px) translateY(153.7664445218706px) rotateX(-144deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-523.2062013197932px) translateY(3.203713998701652e-14px) rotateX(-180deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.244454874933px) translateY(-153.76644452187054px) rotateX(-216deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.4429045449847px) translateY(-248.79933356561165px) rotateX(-252deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.76329677480857px) translateY(-248.79933356561168px) rotateX(-288deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486024px) translateY(-153.76644452187062px) rotateX(-324deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
  <div
    className="roller-wrapper center"
  >
    <div
      className="styles-roller-center"
    />
    <div
      className="roller center"
      style={
        Object {
          "height": "170px",
          "transformOrigin": "0 50% -261.6031006598966px",
          "width": "210px",
        }
      }
    >
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(0px) translateY(0px) rotateX(0deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486021px) translateY(153.76644452187057px) rotateX(-36deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.7632967748085px) translateY(248.79933356561165px) rotateX(-72deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.44290454498463px) translateY(248.79933356561168px) rotateX(-108deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.24445487493296px) translateY(153.7664445218706px) rotateX(-144deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-523.2062013197932px) translateY(3.203713998701652e-14px) rotateX(-180deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.244454874933px) translateY(-153.76644452187054px) rotateX(-216deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.4429045449847px) translateY(-248.79933356561165px) rotateX(-252deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.76329677480857px) translateY(-248.79933356561168px) rotateX(-288deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486024px) translateY(-153.76644452187062px) rotateX(-324deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
  <div
    className="roller-wrapper right"
  >
    <div
      className="styles-roller-right"
    />
    <div
      className="roller right"
      style={
        Object {
          "height": "170px",
          "transformOrigin": "0 50% -261.6031006598966px",
          "width": "210px",
        }
      }
    >
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(0px) translateY(0px) rotateX(0deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486021px) translateY(153.76644452187057px) rotateX(-36deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.7632967748085px) translateY(248.79933356561165px) rotateX(-72deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.44290454498463px) translateY(248.79933356561168px) rotateX(-108deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.24445487493296px) translateY(153.7664445218706px) rotateX(-144deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-523.2062013197932px) translateY(3.203713998701652e-14px) rotateX(-180deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-473.244454874933px) translateY(-153.76644452187054px) rotateX(-216deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-342.4429045449847px) translateY(-248.79933356561165px) rotateX(-252deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-180.76329677480857px) translateY(-248.79933356561168px) rotateX(-288deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        className="slot"
        style={
          Object {
            "transform": "translateZ(-49.96174644486024px) translateY(-153.76644452187062px) rotateX(-324deg)",
          }
        }
      >
        <img
          src="test-file-stub"
        />
      </div>
    </div>
  </div>
</div>
`;
